hopsworks-api/src/main/java/io/hops/hopsworks/api/provenance/ProvenanceCleaner.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * This file is part of Hopsworks
+ * Copyright (C) 2019, Logical Clocks AB. All rights reserved
+ *
+ * Hopsworks is free software: you can redistribute it and/or modify it under the terms of
+ * the GNU Affero General Public License as published by the Free Software Foundation,
+ * either version 3 of the License, or (at your option) any later version.
+ *
+ * Hopsworks is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+ * PURPOSE.  See the GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License along with this program.
+ * If not, see <https://www.gnu.org/licenses/>.
+ */
+package io.hops.hopsworks.api.provenance;
+
+import io.hops.hopsworks.common.dao.hdfs.inode.Inode;
+import io.hops.hopsworks.common.dao.hdfs.inode.InodeFacade;
+import io.hops.hopsworks.common.dao.project.Project;
+import io.hops.hopsworks.common.dao.project.ProjectFacade;
+import io.hops.hopsworks.common.provenance.core.elastic.ProvElasticController;
+import io.hops.hopsworks.common.util.Settings;
+import io.hops.hopsworks.exceptions.ElasticException;
+import io.hops.hopsworks.exceptions.ProvenanceException;
+import io.hops.hopsworks.restutils.RESTCodes;
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.action.admin.indices.delete.DeleteIndexRequest;
+import org.elasticsearch.action.support.master.AcknowledgedResponse;
+import org.elasticsearch.client.indices.GetIndexRequest;
+import org.elasticsearch.client.indices.GetIndexResponse;
+import org.elasticsearch.rest.RestStatus;
+import org.javatuples.Pair;
+
+import javax.ejb.EJB;
+import javax.ejb.Schedule;
+import javax.ejb.Singleton;
+import javax.ejb.Timer;
+import javax.ejb.TransactionAttribute;
+import javax.ejb.TransactionAttributeType;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+@Singleton
+@TransactionAttribute(TransactionAttributeType.NOT_SUPPORTED)
+public class ProvenanceCleaner {
+  private final static Logger LOGGER = Logger.getLogger(ProvenanceCleaner.class.getName());
+  
+  @EJB
+  private Settings settings;
+  @EJB
+  private ProvElasticController client;
+  @EJB
+  private ProjectFacade projectFacade;
+  @EJB
+  private InodeFacade inodeFacade;
+  
+  private String lastIndexChecked = "";
+  
+  // Run once per hour
+  @Schedule(persistent = false, hour = "*", minute = "*")
```

Comment:
This doesn't run once per hour, but every minute of every hour.

Reply:
fixed - switched it to every 4h